site-src/geps/gep-724.md

```diff
@@ -0,0 +1,624 @@
+	// resource type, it SHOULD set the "ResolvedRefs" condition to false for
+	// this listener with the "InvalidRoutesRef" reason.
+	//
+	// Support: Core
+	Kind string `json:"kind"`
+```
+
+#### Added
+```go
+type ListenerRoutes struct {
+  // ...
+  // Kinds specifies the groups and kinds of Routes that are allowed to bind to
+  // this Gateway listener. When unspecified or empty, the only limitation on
+  // the kinds of Routes supported is the Listener protocol. Kind MUST
+  // correspond to kinds of Routes that are compatible with the application
+  // protocol specified in the Listener's Protocol field. If an implementation
+  // does not support or recognize this resource type, it SHOULD set the
+  // "ResolvedRefs" condition to false for this listener with the
+  // "InvalidRoutesRef" reason.
+  //
+  // Support: Core
+  //
+  // +optional
+  Kinds []RouteGroupKind `json:"kinds,omitempty"`
+}
+
+type RouteGroupKind struct {
+  // Group is the group of the Route.
+	//
+	// Support: Core
+	//
+	// +optional
+	// +kubebuilder:default=gateway.networking.k8s.io
+	// +kubebuilder:validation:MaxLength=253
+	Group *string `json:"group,omitempty"`
+	// Kind is the kind of the Route.
+	//
+	// Support: Core
+  //
+  // +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	Kind string `json:"kind"`
+}
+```
+
+### Routes
+
+On Routes, we remove the `RouteGateways` struct and replace it with a list of
+parent references to attach to.
+
+#### Removed
+From Route Specs:
+```go
+  // Gateways defines which Gateways can use this Route.
+  //
+  // +optional
+  // +kubebuilder:default={allow: "SameNamespace"}
+  Gateways *RouteGateways `json:"gateways,omitempty"`
+```
+
+And the structs that references:
+```go
+// RouteGateways defines which Gateways will be able to use a route. If this
+// field results in preventing the selection of a Route by a Gateway, an
+// "Admitted" condition with a status of false must be set for the Gateway on
+// that Route.
+type RouteGateways struct {
+  // Allow indicates which Gateways will be allowed to use this route.
+  // Possible values are:
+  // * All: Gateways in any namespace can use this route.
+  // * FromList: Only Gateways specified in GatewayRefs may use this route.
+  // * SameNamespace: Only Gateways in the same namespace may use this route.
+  //
+  // +optional
+  // +kubebuilder:validation:Enum=All;FromList;SameNamespace
+  // +kubebuilder:default=SameNamespace
+  Allow *GatewayAllowType `json:"allow,omitempty"`
+
+  // GatewayRefs must be specified when Allow is set to "FromList". In that
+  // case, only Gateways referenced in this list will be allowed to use this
+  // route. This field is ignored for other values of "Allow".
+  //
+  // +optional
+  GatewayRefs []GatewayReference `json:"gatewayRefs,omitempty"`
+}
+```
+
+#### Added
+To Route Specs:
+```go
+  // ParentRefs references the parent resources this Route should be attached
```

Comment:
should is lenient and doesn't get the point across.
We want to communicate "anyone can attach to this route ONLY IF they are in this list".
Something along the lines:
```suggestion
  // ParentRefs references the resources that can process/attach/use this Route.
```

Reply:
yeah, agreed, although I would nit that it should be `attach to`. :)

Reply:
Do we think "can attach to" is sufficient here? Not exactly sure how to phrase this, but I think the behavior we want is for these referenced resources to be attached as long as they allow it. Exceptions to that would include conflicts or being referenced from a namespace or kind that isn't allowed. I think we need to be clear that this expresses a desire to attach to the given resource, not just permission for a resource to be attached.